specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/ContainerApps.json

```diff
@@ -551,6 +605,38 @@
           "type": "boolean"
         }
       }
+    },
+    "SecretsCollection": {
+      "description": "Container App Secrets Collection ARM resource.",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ContainerAppSecret"
+          }
+        }
+      }
+    },
+    "ContainerAppSecret": {
+      "description": "Container App Secret.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Secret Name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "value": {
```

Comment:
>value [](http://example.com/codeflow?start=9&length=5)

This property needs to be tagged with the x-ms-secret extension. Ref: [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret)

Reply:
updated

Reply:
All secrets need the x-ms-secret extension and are not allowed in GET responses. If the secrets need to be retrieved, it has to be done via a separate POST action (if you get a linter error for that, you will need to ignore it and have your PR assignee bypass-- there is currently a buggy linter rule).

Reply:
@mentat9 - this is a POST action API.

Reply:
@leni-msft could you help bypass the error?

Reply:
secrets are allowed to be in POST response